hedged.go

```diff
@@ -58,11 +64,15 @@ func (ht *hedgedTransport) RoundTrip(req *http.Request) (*http.Response, error)
 	resultCh := make(chan *http.Response, ht.upto)
 	errorCh := make(chan error, ht.upto)
 
+	cancelContext := ht.cancelContext
 	for sent := 0; len(errOverall.Errors) < ht.upto; sent++ {
 		if sent < ht.upto {
 			runInPool(func() {
-				req, cancel := reqWithCtx(req, mainCtx)
-				defer cancel()
+				if cancelContext {
+					var cancel func()
+					req, cancel = reqWithCtx(req, mainCtx)
```

Comment:
I wonder, how bad it would be if we will remove this line at all? 👀 

I mean let's not create a context here at all(yes-yes!) 'cause all the requests here will be canceled by the original context tied to the original request.

And if this will 'cause performance degradation we will add this flag. Objections?